File: C++/0287. 字符串匹配.cpp
// we have defined the necessary header files here for this problem.
// If additional header files are needed in your program, please import here.
#include<algorithm>
int main()
{  
  // please define the C++ input here. For example: int a,b; cin>>a>>b;;  
  // please finish the function body here.  
  // please define the C++ output here. For example:cout<<____<<endl; 
    string word;
    cin>>word;
    transform(word.begin(), word.end(), word.begin(), ::tolower); 
    int n;
    cin>>n;
    for(int i=0;i<n;++i) {
        string sentence;
        string temp;
        cin>>sentence;
        temp = sentence;
        transform(temp.begin(), temp.end(), temp.begin(), ::tolower);
        if(temp.find(word) != string::npos) {
            cout<<sentence<<endl;
        }
    }
    return 0;
}

File: Python/0287. 字符串匹配.py
# If you need to import additional packages or classes, please import here.

def func():
    # please define the python3 input here. 
    # For example: a,b = map(int, input().strip().split())
    # please finish the function body here.
    # please define the python3 output here. For example: print().
    word = input()
    word = word.lower()
    n = int(input())
    for _ in range(n):
        sentence = input()
        temp = sentence
        temp = temp.lower()
        if word in temp:
            print(sentence)
    
if __name__ == "__main__":
    func()
